relay-server/src/envelope.rs

```diff
@@ -554,6 +554,14 @@ pub struct ItemHeaders {
     #[serde(default, skip_serializing_if = "is_false")]
     metrics_extracted: bool,
 
+    /// Whether or not a transaction has been extracted from a segment span.
+    #[serde(default, skip_serializing_if = "is_false")]
+    transaction_extracted: bool,
+
+    /// Whether or not spans have been extracted from a transaction.
+    #[serde(default, skip_serializing_if = "is_false")]
+    spans_extracted: bool,
```

Comment:
With these two flags, we can prevent conversion between events and spans from going in circles.